tools/wptrunner/wptrunner/testrunner.py

```diff
@@ -571,21 +571,30 @@ def run_test(self):
         self.timer.start()
 
     def _timeout(self):
+        # This is executed in a different thread (threading.Timer).
         self.logger.info("Got timeout in harness")
         test = self.state.test
-        self.test_ended(test,
-                        (test.result_cls("EXTERNAL-TIMEOUT",
-                                         "TestRunner hit external timeout "
-                                         "(this may indicate a hang)"), []))
+        self.command_queue.put((
```

Comment:
Oh per the CI this seems to be not quite correct.

Reply:
Hmm, why did this pass all the tests last time (before I changed asserts to errors)...

Anyway, I added a new method to this class called `inject_message` as a counterpart of `send_messsage` (`inject_message` sends a message to the queue where the TestRunner normally receive messages from) to avoid the nasty nested tuples.